python/kiss_icp/pipeline.py

```diff
@@ -97,8 +97,8 @@ def _run_pipeline(self):
             raw_frame, timestamps = self._next(idx)
             start_time = time.perf_counter_ns()
             source, keypoints = self.odometry.register_frame(raw_frame, timestamps)
-            self.poses.append(self.odometry.last_pose)
-            self.times.append(time.perf_counter_ns() - start_time)
+            self.poses[idx - self._first] = self.odometry.last_pose
+            self.times[idx - self._first] = time.perf_counter_ns() - start_time
             self.visualizer.update(source, keypoints, self.odometry.local_map, self.poses[-1])
```

Comment:
This currently breaks the visualizer, because the last pose is initialized with zeros, we can use `self.odometry.last_pose` here